src/server/gc_worker/compaction_filter.rs

```diff
@@ -103,6 +106,23 @@ impl CompactionFilterFactory for WriteCompactionFilterFactory {
             return std::ptr::null_mut();
         }
 
+        let mut mvcc_props = MvccProperties::new();
+        for i in 0..context.file_numbers().len() {
+            let table_props = context.table_properties(i);
+            let user_props = unsafe {
+                &*(table_props.user_collected_properties() as *const _
+                    as *const RocksUserCollectedPropertiesNoRc)
+            };
+            if let Ok(props) = RocksMvccProperties::decode(user_props) {
+                mvcc_props.add(&props);
+            }
+        }
+
+        let ratio_threshold = gc_context.cfg_tracker.value().ratio_threshold;
+        if !check_need_gc(safe_point.into(), ratio_threshold, mvcc_props) {
```

Comment:
Shall it be too high to decide one gc job? If there are some files which has a log of `keys` need to be gc, but they are overlap with many files in higher level so that the total ratio can not reach 10%.

Reply:
It's right. But seems we don't have better solutions?

Reply:
If total ratio > 0.1 or single sst ratio > 0.3 { do a compaction filter job }